app/move/controllers/create/assessment.js

```diff
@@ -1,15 +1,47 @@
-const { flatten, values } = require('lodash')
+const { forEach, flatten, values, mapValues } = require('lodash')
 
 const CreateBaseController = require('./base')
 const fieldHelpers = require('../../../../common/helpers/field')
+const createFields = require('../../fields/create')
+const referenceDataService = require('../../../../common/services/reference-data')
 
 class AssessmentController extends CreateBaseController {
   async configure(req, res, next) {
     try {
-      const { fields } = req.form.options
-      req.form.options.fields = await fieldHelpers.populateAssessmentQuestions(
-        fields
+      const { fields, assessmentCategory } = req.form.options
+      const questions = await referenceDataService.getAssessmentQuestions(
+        assessmentCategory
       )
+
+      const implicitField = createFields.assessmentCategory(assessmentCategory)
+      implicitField.items = questions
+        .filter(question => {
+          const key = question.key
+          const field = fields[key]
+
+          return fields[key] && !field.explicit
+        })
+        .map(fieldHelpers.mapAssessmentQuestionToConditionalField)
+        .map(fieldHelpers.mapAssessmentQuestionToTranslation)
+        .map(fieldHelpers.mapReferenceDataToOption)
+
+      req.form.options.fields[assessmentCategory] = implicitField
+
+      req.form.options.fields = mapValues(
+        req.form.options.fields,
+        createFields.appendDependent.bind(null, questions, assessmentCategory)
+      )
+
+      req.form.options.fields = forEach(
```

Comment:
I wonder if we can also combine this loop with the map. At the moment we're having this controller loop over all fields twice. Could the map call a method within it to append the explicit fields?

Reply:
Whilst I'd love to get rid of the additional loop, I don't quite see how this could work. In fact, we do mutate `req.form.options.fields` -- this is how we append the additional field. I mean, we rely on a side effect. At least in this way it's fairly isolated...

Reply:
But we could move the mutation into a separate method that creates a copy of the original fields and appends to that, then returns it.

Trying to apply more of a functional approach where methods return a value rather than updating things too much of their scope.

Granted, we need to mutate the fields object within the controller, but this is a slight limitation of the Form Wizard. What we can try and do is reduce the complexity of these configure methods to call our helpers which are written in a more functional way. That way our helpers are more easily tested as they take an input and we can assert the output.

And our tests for the controllers which test mutation stay as light as possible too.

Reply:
Ok, makes sense.